### PR TITLE
Features/civ 5574 claim intent check answers

### PR DESCRIPTION
**Pull request by pascalkankonde**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-5574

### Change description ###

The scope of this ticket is to develop the 'Check your answers before submitting' screen in the scenario where the Defendant has fully admitted and offered to pay by a set date. This screen occurs after the citizen has clicked on the "Notify us about the debt respite scheme" option on the latest update section.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/services/features/breathingSpace/debtRespiteScheme/buildDebtRespiteSection.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  BREATHING_SPACE_CHECK_ANSWERS_URL,
+  BREATHING_SPACE_RESPITE_END_DATE_URL,
+  BREATHING_SPACE_RESPITE_START_DATE_URL,
+  BREATHING_SPACE_RESPITE_TYPE_URL,
+} from '../../../../routes/urls';
+import {constructResponseUrlWithIdParams} from '../../../../common/utils/urlFormatter';
+import {BreathingSpace} from 'models/breathingSpace';
+import {formatDateToFullDate} from 'common/utils/dateUtils';
+
+const changeLabel = (lang: string | unknown): string => t('COMMON.BUTTONS.CHANGE', {lng: getLng(lang)});
```
Comment (laperla11) [inline src/main/services/features/breathingSpace/debtRespiteScheme/buildDebtRespiteSection.ts:15]: You can import this function from `src/main/common/utils/checkYourAnswer/changeButton.ts`

File: src/main/common/models/breathingSpace/debtRespiteReferenceNumber.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export class DebtRespiteReferenceNumber {
```
Comment (laperla11) [inline src/main/common/models/breathingSpace/debtRespiteReferenceNumber.ts:1]: Where do you use this model?

File: src/main/routes/features/claim/breathingSpace/checkAnswersController.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import {NextFunction, Response, Router} from 'express';
+import {BREATHING_SPACE_CHECK_ANSWERS_URL, DASHBOARD_CLAIMANT_URL} from '../../../urls';
+import {getSummarySections} from '../../../../services/features/breathingSpace/checkAnswersService';
+import {constructResponseUrlWithIdParams} from '../../../../common/utils/urlFormatter';
+import {AppRequest} from '../../../../common/models/AppRequest';
+import {getBreathingSpace} from 'services/features/breathingSpace/breathingSpaceService';
+import {BreathingSpace} from 'models/breathingSpace';
+//import {getBreathingSpace} from 'services/features/breathingSpace/breathingSpaceService';
```
Comment (laperla11) [inline src/main/routes/features/claim/breathingSpace/checkAnswersController.ts]: remove the code if not needed

File: src/main/routes/urls.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 const BASE_CASE_URL = '/case/:id';
 const BASE_CLAIM_URL = '/claim';
+//const BASE_BREATHING_SPACE_URL = `${BASE_CASE_URL}/breathing-space`;
```
Comment (laperla11) [inline src/main/routes/urls.ts]: same here

**Review by laperla11 — CHANGES_REQUESTED** [review]

File: src/main/common/models/breathingSpace/debtRespiteReferenceNumber.ts (hunk 2 above)
Comment (pascalkankonde) [reply src/main/common/models/breathingSpace/debtRespiteReferenceNumber.ts:1]: it was duplicated import, thank you for spotting that commented line. I have removed it.

File: src/main/services/features/breathingSpace/debtRespiteScheme/buildDebtRespiteSection.ts (hunk 1 above)
Comment (pascalkankonde) [reply src/main/services/features/breathingSpace/debtRespiteScheme/buildDebtRespiteSection.ts:15]: It was duplicated import, thank you for spotting that commented line. I have removed it.

File: src/main/routes/features/claim/breathingSpace/checkAnswersController.ts (hunk 3 above)
Comment (pascalkankonde) [reply src/main/routes/features/claim/breathingSpace/checkAnswersController.ts]: Yes, it was duplication 6 and 8, Removed now

File: src/main/routes/urls.ts (hunk 4 above)
Comment (pascalkankonde) [reply src/main/routes/urls.ts]: duplication import commented, removed

File: src/main/services/features/breathingSpace/debtRespiteScheme/buildDebtRespiteSection.ts (hunk 1 above)
Comment (pascalkankonde) [reply src/main/services/features/breathingSpace/debtRespiteScheme/buildDebtRespiteSection.ts:15]: Done

File: src/main/common/models/breathingSpace/debtRespiteReferenceNumber.ts (hunk 2 above)
Comment (pascalkankonde) [reply src/main/common/models/breathingSpace/debtRespiteReferenceNumber.ts:1]: This has been used in  src/main/services/features/breathingSpace/debtRespiteScheme/buildDebtRespiteSection.ts  line 23 to get the referenceNumber

**Review by laperla11 — APPROVED** [review]

File: src/main/modules/i18n/locales/en.json [hunk 5]
```diff
@@ -608,6 +609,18 @@
       "LATEST_UPDATE": "Latest update",
       "VIEW_CLAIM": "View claim"
     },
+    "CLAIMANT_DEBT_RESPITE_CHECK_ANSWERS": {
```
Comment (laarmada) [inline src/main/modules/i18n/locales/en.json:612]: we have miss some translations...

File: src/main/services/features/breathingSpace/checkAnswersService.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import {SummarySections} from '../../../common/models/summaryList/summarySections';
+import {buildDebtRespiteSection} from './debtRespiteScheme/buildDebtRespiteSection';
+import {BreathingSpace} from 'models/breathingSpace';
+
+const buildSummarySections = (breathingSpace: BreathingSpace, claimId: string, lang: string | unknown): SummarySections => {
+
+  return {
+    sections: [
```
Comment (laarmada) [inline src/main/services/features/breathingSpace/checkAnswersService.ts:8]: where is the test?

**Review by laarmada — CHANGES_REQUESTED** [review]

File: src/main/modules/i18n/locales/en.json (hunk 5 above)
Comment (pascalkankonde) [reply src/main/modules/i18n/locales/en.json:612]: CLAIMANT_DEBT_RESPITE_CHECK_ANSWERS  is fine, I can see the Gap, maybe if related tasks are merge will solve the issue.

File: src/main/services/features/breathingSpace/checkAnswersService.ts (hunk 6 above)
Comment (pascalkankonde) [reply src/main/services/features/breathingSpace/checkAnswersService.ts:8]: I have test for that implantation in this files src/test/unit/services/features/breathingSpace/buildDebtRespiteSection.test.ts   there is  buildSummarySections implation which is called in checkAnswersService

**Review by laarmada — APPROVED** [review]

**Comment by laarmada** [comment]
I think you will have a sonar issue..

**Pull request closed without merging**